fastqfil/additional_scripts/procedures_with_dna_and_rna.py

```diff
@@ -0,0 +1,76 @@
+# Verify the existence of the requested procedure
+def check_of_procedure(procedure_name):
+    return procedure_name in [
+        "transcribe",
+        "reverse",
+        "complement",
+        "reverse_complement",
+    ]
+
+
+# To return a complementary sequence
+def complement(chain):
+    complementary_chain = ""
+    for nucleotide in chain:
+        if nucleotide == "A":
+            complementary_chain += "T"
+        elif nucleotide == "a":
+            complementary_chain += "t"
+        elif nucleotide == "U":
+            complementary_chain += "A"
+        elif nucleotide == "u":
+            complementary_chain += "a"
+        elif nucleotide == "T":
+            complementary_chain += "A"
+        elif nucleotide == "t":
+            complementary_chain += "a"
+        elif nucleotide == "C":
+            complementary_chain += "G"
+        elif nucleotide == "c":
+            complementary_chain += "g"
+        elif nucleotide == "G":
+            complementary_chain += "C"
+        elif nucleotide == "g":
+            complementary_chain += "c"

```

Comment:
Тяжеловесную структуру условных операторов можно избежать, если определить словарь и обращаться за нужным соответствующим значением к нему.

на глобальном для скрипта уровне можно определить словарь

```python
complement_dict = {
    "A": "T",
    "a": "t",
    "U": "A",
    "u": "a",
    "T": "A",
    "t": "a",
    "C": "G",
    "c": "g",
    "G": "C",
    "g": "c",
}
```

А в этой функции упростить получение нужного нуклеотида
```suggestion
        complementary_chain += complement_dict[nucleotide]
```

Reply:
Абсолютно верно!